src/interface/src/app/top-bar/top-bar.component.ts

```diff
@@ -44,8 +45,10 @@ export class TopBarComponent implements OnInit, OnDestroy {
             : user.username
             ? user.username
             : '';
```

Comment:
It looks like it's possible for the displayName to be an empty string. Just confirming, does the avatar just show a blank circle if that's the case? (imo that's probably fine, but if it's not too much effort to code it may be worth catching that edge case by showing the account icon instead)

Reply:
sounds good!

Reply:
Honestly, unsure (this is legacy code) — I'm wondering if it's to account for if the name/username gets corrupted/can't get retrieved for some reason? At least the name is definitely required during account creation

Reply:
We could have a different default username instead of a blank string (like 'User'?). That may be something we'd need to check with Michele/Scott (Or see if this is even a situation we need to have a backup for in the first place)

Reply:
updated to check for displayname

Reply:
Amazing, ty!

Reply:
Would you actually mind adding in a comment documenting this logic? (displayName is dependent on first name and then username, and also if neither of those are available then the avatar will be a default account icon)